internal/service/advancedclustertpf/validate_schema.go

```diff
@@ -82,3 +83,25 @@ func performValidation(ctx context.Context, state *tfsdk.State, plan *tfsdk.Plan
 		diags.AddError(fmt.Sprintf("`%s` %s", validationPath, v.Description(ctx)), "")
 	}
 }
+
+type RegionSpecPriorityOrderDecreasingValidator struct{}
+
+func (v RegionSpecPriorityOrderDecreasingValidator) Description(ctx context.Context) string {
+	return v.MarkdownDescription(ctx)
+}
+func (v RegionSpecPriorityOrderDecreasingValidator) MarkdownDescription(_ context.Context) string {
+	return "must be a list with priority in descending order"
+}
+func (v RegionSpecPriorityOrderDecreasingValidator) ValidateList(ctx context.Context, req validator.ListRequest, resp *validator.ListResponse) {
+	diags := &resp.Diagnostics
+	regionConfigs := newCloudRegionConfig20240805(ctx, req.ConfigValue, diags)
+	if diags.HasError() {
+		return
+	}
+	configs := *regionConfigs
+	for i := range configs[:len(configs)-1] {
```

Comment:
```suggestion
	for i := len(configs)-1
```

in later Go versions you can use range in an int,  this is equivalent in this case